### PR TITLE
Add support for Plone 5.1.

### PR DESCRIPTION
**Pull request by mbaechtold**

### CODE REVIEW

File: ftw/copymovepatches/tests/test_catalog_fixes_view.py [hunk 1]
```diff
@@ -32,6 +34,7 @@ def test_rebuild_broken_uid_index(self, browser):
             browser.css('.uid-index-state-check').first.text)
 
     @browsing
+    @skipIf(IS_PLONE_5, 'We no longer create AT content within DX.')
```
Comment (mbaechtold) [inline ftw/copymovepatches/tests/test_catalog_fixes_view.py]: @jone, is this assumption correct? If yes, does it mean that the brain metadata won't be broken at all?

File: ftw/copymovepatches/testing.py [hunk 2]
```diff
@@ -23,11 +25,16 @@ def setUpZope(self, app, configurationContext):
...
-        z2.installProduct(app, 'collective.indexing')
+        if not IS_PLONE_5:
+            z2.installProduct(app, 'collective.indexing')
 
     def setUpPloneSite(self, portal):
         applyProfile(portal, 'ftw.copymovepatches.tests:default')
 
+        if IS_PLONE_5:
+            applyProfile(portal, 'plone.app.contenttypes:default')
+            register_dx_content_builders(force=True)
```
Comment (jone) [inline ftw/copymovepatches/testing.py]: Is this necessary?
If I remember correctly, with Plone 5 we should always have DX builders.
Using this function would then only make sense when running Plone 4.

File: ftw/copymovepatches/tests/test_catalog_fixes_view.py (hunk 1 above)
Comment (jone) [reply ftw/copymovepatches/tests/test_catalog_fixes_view.py]: It is true that the specific use case described in the readme (AT in DX) does not happen when we have no AT anymore.
I'm not sure though if there are any use cases which may cause a similar problem.
If possible, I'd try to keep the test with DX.

What happens when running the test with DX only? Is there a problem with changing the UID without indexing?

File: ftw/copymovepatches/testing.py (hunk 2 above)
Comment (mbaechtold) [reply ftw/copymovepatches/testing.py]: Right, we can omit that.

File: ftw/copymovepatches/tests/test_catalog_fixes_view.py [hunk 3]
```diff
@@ -68,6 +70,16 @@ def break_object(self, obj):
...
         transaction.commit()
+
+    def _change_uid(self, obj):
+
+        # Change UID for Archetypes:
+        if hasattr(obj, config.UUID_ATTR):
+            setattr(obj, config.UUID_ATTR, getUtility(IUUIDGenerator)())
+
+        # Change UID for Dexterity:
+        if hasattr(obj, ATTRIBUTE_NAME):
```
Comment (mbaechtold) [inline ftw/copymovepatches/tests/test_catalog_fixes_view.py]: @jone, is this the right attribute for DX content?

File: ftw/copymovepatches/tests/test_catalog_fixes_view.py (hunk 1 above)
Comment (mbaechtold) [reply ftw/copymovepatches/tests/test_catalog_fixes_view.py]: The problem was the code which changed the uid of the object in the test: it only worked for AT. I have added a DX version, see [my other comment](https://github.com/4teamwork/ftw.copymovepatches/pull/11/files#r160102408).

File: ftw/copymovepatches/tests/test_catalog_fixes_view.py (hunk 3 above)
Comment (jone) [reply ftw/copymovepatches/tests/test_catalog_fixes_view.py]: I think so; if it works, it is 😉

**Pull request merged**